### PR TITLE
Fix #2561 make Missing option on terms aggregation generic

### LINKED ISSUE #2561: .Missing() for Term-aggregation of int-field
**NEST/Elasticsearch.Net version**: 5.0.1

**Elasticsearch version**: 5.1.2

**Description of the problem including expected versus actual behavior**:

When specifying `.Missing()` for Term-aggregation of integer-fields elastic reponds with a confusing error. 
It appears elastic 5 is stricter on types for missing-values in aggregations. `"missing":-1` works, but not `"missing":"-1"`.

**Steps to reproduce**:

```c
void Main()
{
	var client = new ElasticClient(new Uri("http://localhost:9200"));
	
	client.CreateIndex("fooindex", c => c.Settings(s => s.NumberOfShards(1).NumberOfReplicas(0))
		.Mappings(ms => ms.Map<Foo>(m => m.AutoMap()))
	);
	
	var resp = client.Search<Foo>(s => s
		.Index("fooindex")
		.Aggregations(a => a
			.Terms("a", t => t.Field(b => b.SomeInt)
				.Missing("-1") // need overload that takes int?
			)
		)
	);
	
	// writes: Field [someInt] of type [integer] does not support custom time zones
	Console.WriteLine(resp.ServerError.Error.RootCause.First().Reason); 
}

public class Foo
{
	public int Id { get; set; }
	public int SomeInt { get; set; }
}
```

### PR DESCRIPTION
**Pull request by gmarz**

### CODE REVIEW

File: src/Nest/Aggregations/Bucket/Terms/TermsAggregation.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public interface ITermsAggregation : IBucketAggregation
 		TermsAggregationCollectMode? CollectMode { get; set; }
 
 		[JsonProperty("missing")]
-		string Missing { get; set; }
+		object Missing { get; set; }
```
Comment (Mpdreamz) [inline src/Nest/Aggregations/Bucket/Terms/TermsAggregation.cs]: Why did you not opt for `TFieldType`, not crazy about the boxing here.

Comment (gmarz) [inline src/Nest/Aggregations/Bucket/Terms/TermsAggregation.cs]: Me either, but we'd have to make `AggregationContainer<TFieldType>` as well which opens up a whole can of worms, and I'm not sure we really should.

**Comment by Mpdreamz** [comment]
@gmarz added a commit with a way to prevent boxing, including an integration/unit tests for the missing property sending `-1`.

File: src/Nest/Aggregations/AggregationContainer.cs [hunk 2]
```diff
@@ -500,7 +500,11 @@ public class AggregationContainerDescriptor<T> : DescriptorBase<AggregationConta
 			_SetInnerAggregation(name, selector, (a, d) => a.Sum = d);
 
 		public AggregationContainerDescriptor<T> Terms(string name,
-			Func<TermsAggregationDescriptor<T>, ITermsAggregation> selector) =>
+			Func<TermsAggregationDescriptor<T, string>, ITermsAggregation<string>> selector) =>
+			_SetInnerAggregation(name, selector, (a, d) => a.Terms = d);
+
+		public AggregationContainerDescriptor<T> Terms<TMissingType>(string name,
+			Func<TermsAggregationDescriptor<T, TMissingType>, ITermsAggregation<TMissingType>> selector) =>
```
Comment (gmarz) [inline src/Nest/Aggregations/AggregationContainer.cs]: I would rename `TMissingType` => `TFieldType`

**Comment by gmarz** [comment]
LGTM 👍 Just a minor nitpick

**Comment by gmarz** [comment]
Much better BTW :)

**Comment by Mpdreamz** [comment]
rebased and amended @gmarz 👍

**Pull request merged**